### PR TITLE
For #8108 - Add BrowserToolbar option to remove url from TopSites

### PR DESCRIPTION
**Pull request by Alexandru2909**
Added remove top site option to BrowserToolbar, when current url is a pinned site. When clicked, the url is removed from pinned sites and the menu item goes back to "Add to top sites".

remove

removesnack

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
@mcarare Tagged you since I recall you did something similar in Focus.

File: app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt [hunk 1]
```diff
@@ -632,7 +632,7 @@ sealed class Event {
         enum class Item {
             SETTINGS, HELP, DESKTOP_VIEW_ON, DESKTOP_VIEW_OFF, FIND_IN_PAGE, NEW_TAB,
             NEW_PRIVATE_TAB, SHARE, BACK, FORWARD, RELOAD, STOP, OPEN_IN_FENIX,
-            SAVE_TO_COLLECTION, ADD_TO_TOP_SITES, ADD_TO_HOMESCREEN, QUIT, READER_MODE_ON,
+            SAVE_TO_COLLECTION, ADD_TO_TOP_SITES, REMOVE_FROM_TOP_SITES, ADD_TO_HOMESCREEN, QUIT, READER_MODE_ON,
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/components/metrics/Event.kt:635]: This should also be added to the description of extra keys in a `browser_menu_action` in the `metrics.yaml` file.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 2]
```diff
@@ -48,6 +50,7 @@ import org.mozilla.fenix.utils.BrowsersCache
  * @param store reference to the application's [BrowserStore].
  * @param hasAccountProblem If true, there was a problem signing into the Firefox account.
  * @param shouldReverseItems If true, reverse the menu items.
+ * @param pinnedSiteStorage Used to check if the current url is pinned.
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:53]: nit: "is a pined top site"

File: app/src/test/java/org/mozilla/fenix/components/toolbar/DefaultBrowserToolbarMenuControllerTest.kt [hunk 3]
```diff
@@ -623,7 +649,7 @@ class DefaultBrowserToolbarMenuControllerTest {
         activity: HomeActivity = this.activity,
         customTabSessionId: String? = null,
         findInPageLauncher: () -> Unit = { },
-        bookmarkTapped: (String, String) -> Unit = { _, _ -> }
+        bookmarkTapped: (String, String) -> Unit = { _, _ -> },
```
Comment (mcarare) [inline app/src/test/java/org/mozilla/fenix/components/toolbar/DefaultBrowserToolbarMenuControllerTest.kt:652]: remove the trailing comma.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1166,7 +1166,7 @@
     <!-- Text shown in snackbar to undo deleting a tab, top site or collection -->
     <string name="snackbar_deleted_undo">UNDO</string>
     <!-- Text shown in snackbar when user removes a top site -->
-    <string name="snackbar_top_site_removed" moz:removedIn="96" tools:ignore="UnusedResources">Site removed</string>
+    <string name="snackbar_top_site_removed" tools:ignore="UnusedResources">Site removed</string>
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply app/src/main/res/values/strings.xml:1169]: This should be fine, but I also see we kept `tools:ignore="UnusedResources"`. Is the string being used?

[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/main/res/values/strings.xml:1169]: Yes, that should also be removed. cc @Alexandru2909

**Review by mcarare — APPROVED** [review]
👍

**Pull request merged**